node/device/node-red/azureiothub/azureiothub.js

```diff
@@ -0,0 +1,162 @@
+        mqtt: require('azure-iot-device-mqtt').Mqtt,
+        http: require('azure-iot-device-http').Http,
+        amqpWs: require('azure-iot-device-amqp-ws').AmqpWs
+    };
+
+    var Message = require('azure-iot-device').Message;
+
+    var client = null;
+    var clientConnectionString = "";
+    var clientProtocol = "";
+    var pendingMessage = null;
+    var node = null;
+    var nodeConfig = null;
+
+    var statusEnum = {
+        disconnected: { color: "red", text: "Disconnected" },
+        connected: { color: "green", text: "Connected" },
+        sent: { color: "blue", text: "Sent message" },
+        received: { color: "yellow", text: "Received" },
+        error: { color: "grey", text: "Error" }
+    };
+
+    var setStatus = function (status) {
+        node.status({ fill: status.color, shape: "dot", text: status.text });
+    }
+
+    var sendData = function (data) {
+        node.log('Sending Message to Azure IoT Hub :\n   Payload: ' + data.toString());
+        // Create a message and send it to the IoT Hub every second
+        var message = new Message(data);
+        client.sendEvent(message, function (err, res) {
+            if (err) {
+                node.error('Error while trying to send message:' + err.toString());
+                setStatus(statusEnum.error);
+            } else {
+                node.log('Message sent.');
+                setStatus(statusEnum.sent);
+            }
+        });
+    };
+
+    var sendMessageToIoTHub = function (message, reconnect) {
+        if ((client == null) || reconnect) {
+            node.log('Connection to IoT Hub not established or configuration changed. Reconnecting.');
+            // Store the message data to send it once the connection is established
+            pendingMessage = message;
+            // Update the connection string
+            clientConnectionString = node.credentials.connectionstring;
+            // update the protocol
+            clientProtocol = nodeConfig.protocol;
+
+            // If client was previously connected, disconnect first
+            if (client != null)
+                disconnectFromIoTHub();
+
+            // Connect the IoT Hub
+            connectToIoTHub();
+        } else {
+            sendData(message);
+        }
+    };
+
+    var connectToIoTHub = function () {
+        node.log('Connecting to Azure IoT Hub:\n   Protocol: ' + clientProtocol + '\n   Connection string :' + clientConnectionString);
+        client = Client.fromConnectionString(clientConnectionString, Protocols[clientProtocol]);
+        client.open(function (err) {
+            if (err) {
+                node.error('Could not connect: ' + err.message);
+                setStatus(statusEnum.disconnected);
+            } else {
+                node.log('Connected to Azure IoT Hub.');
+                setStatus(statusEnum.connected);
+
+                // Check if a message is pending and send it 
+                if (pendingMessage != null) {
```

Comment:
instead of using pendingMessage as a module variable you could just pass it as an argument. would work better if for some reason there are rapid subsequent calls to connectIoTHub (avoid squashing the previous pending message)